### PR TITLE
Add documentation for TMA notebook

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Addresses the second half of #188. The TMA notebook contains a lot good, thorough documentation, however it should be formatted per @HPiyadasa requests for the tiled regions notebook.

Additionally, this PR addresses the minor issue of the save annotation not refreshing with the updated `matplotlib`.

**How did you implement your changes**

Reformat documentation with headings and bolded text to indicate importance. Add `ax.canvas.figure.draw()` to the save helper function of `tma_interactive_remap` to ensure `save_ann` draws immediately after `Save mapping` is clicked, and not with a single-event delay.

**Remaining issues**

I've left much of the documentation in the beginning as reference, but it will probably be removed. @HPiyadasa let me know what you want me to do with it on review.

### CODE REVIEW

**Comment by alex-l-kong** [comment]
@HPiyadasa here's a rough draft of the documentation for the TMA notebook; tried to maintain a similar structure to the tiled region notebook. Per the PR description, I've left some documentation Noah originally wrote at the top, let me know what you want to do with them.

**Comment by alex-l-kong** [comment]
@ngreenwald @HPiyadasa upon further review, I'm going to end up reverting most of the changes back to how the original notebook was formatted.

I recommend the following changes:

1. Section 2: separate out `tma_num_row` and `tma_num_col` definition into a separate cell, so the user knows they need to set this
2. Section 2: move loading of `slide_data` to the section 3, in the same cell just before starting up the GUI
3. Section 2: combine `auto_fov_regions` generation and `manual_fov_regions` loading/validating into one cell
4. Section 3: add a GIF demonstrating Zoom feature, and a GIF demonstrating both the toggle menu (Manually-defined FOV) and the reassignment menu (Automatically-generated FOV)

We can meet sometime this week to discuss if anything is unclear.

File: templates/2_create_tma_mibi_run.ipynb [hunk 1]
```diff
@@ -30,14 +30,14 @@
     "This notebook automatically checks the names assigned to the cores on a TMA. In order to get the most benefit out of the notebook, make sure that you've named your FOVs appropriately. The expected format is RNCM, where N is the row and M is the column of the TMA. For example, a core on the third row and second column would be R3C2, and one on the 7th row and first column would be R7C1.  \n",
```
Comment (ngreenwald) [inline templates/2_create_tma_mibi_run.ipynb:30]: <p>Looks like some, but not all, of the references to <code>tma_name</code> were changed to <code>tma_prefix</code> instead. Please make sure everything in here is consistent  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/toffy/pull/207/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Looks good, just need to update the references to the right prefix variable

**Comment by ngreenwald** [comment]
looks good, will wait till @HPiyadasa takes a look before merging in

File: templates/2_create_tma_mibi_run.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-l-kong) [reply templates/2_create_tma_mibi_run.ipynb:30]: @HPiyadasa yeah we can do that, just to clarify we run distance verification in microns.

@ngreenwald how does this sound:

set to a positive value (in microns) to notify of FOV mappings at a distance greater than this value, sorted by decreasing distance. Adjust based on how tightly or sparsely packed the TMA cores are. Set to `None` to bypass.

[the remark this answers is not in the document]
Comment (ngreenwald) [reply templates/2_create_tma_mibi_run.ipynb:30]: @HPiyadasa have we ever had to adjust this? Maybe we should hide this as a setting?

[the remark this answers is not in the document]
Comment (HPiyadasa) [reply templates/2_create_tma_mibi_run.ipynb:30]: <p>@ngreenwald We have this TMA with 180 cores and they are all 600 micron diameter and its fairly packed. I would be curious if we will have to adjust this to get that to work?  </p>

[the remark this answers is not in the document]
Comment (ngreenwald) [reply templates/2_create_tma_mibi_run.ipynb:30]: <p>Hmm not sure. Lets keep it then, we can go with Alex's proposed changes</p>

**Comment by alex-l-kong** [comment]
@ngreenwald addressed @HPiyadasa's comment, does this look good to go now?

**Review by HPiyadasa — APPROVED** [review]

**Comment by alex-l-kong** [comment]
@ngreenwald looks like documentation is good to go, any final comments?

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**